presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -140,9 +140,23 @@ protected Node visitShowStats(ShowStats node, Void context)
                 throw rewriteAccessDeniedException(e);
             }
 
+            Table table = getTable(node, specification);
+            QualifiedObjectName tableName = createQualifiedObjectName(session, node, table.getName());
+            TableHandle tableHandle = metadata.getTableHandle(session, tableName)
+                    .orElseThrow(() -> semanticException(TABLE_NOT_FOUND, node, "Table '%s' not found", table.getName()));
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+            Set<String> columnNames = extractStatsColumns(tableMetadata, specification.getSelect().getSelectItems());
+
+            try {
+                accessControl.checkCanSelectFromColumns(session.toSecurityContext(), tableName, columnNames);
```

Comment:
> Do validation after access control in ShowStats
> 
> Validation potentially could leak information about column masks and row
> filters used during the planning (done within ShowStatsRewrite).


- Can we have a test covering precisely this?
- should we ask AC whether there are any masks?

Reply:
> should we ask AC whether there are any masks?

I do that in next commit.

Reply:
> Validation potentially could leak information about column masks and row

I meant that if column mask is a function the user might get an error with information showing that function.

Reply:
so maybe squash the two commits

Reply:
So I did